templates/support/SecurityClient.swift.hbs

```diff
@@ -0,0 +1,361 @@
+//  
+//  {{>frag/generatedBy}}
+//
+
+import Foundation
+
+public protocol SecurityClient {
+    /** An API request received a 401 response */
+    func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws
+    
+    /** An authorization header to the request */
+    func authorize(securityScheme: SecurityScheme, scopes: [String]?, request: URLRequest) async throws -> URLRequest
+}
+
+/** Delegates security requiements to the appropriate SecurityClient. */
+public class SecurityClientController: SecurityClient {
```

Comment:
I'm feeling like it's cool but a little confusing that we've got `SecurityClient` as the protocol for this multiplexer and for our individual implementations... it makes sense technically but I think it's a little confusing: "Do I put my security client into the `Configuration` directly or..."

Perhaps we should _default_ the `securityClient` on `Configuration` to a `SecurityClientController` so it's a little more obvious that you add your own `SecurityClient` to the controller, not directly to the `Configuration`.

Reply:
I'm not sure about this. It seems like we could modify the constructor for Configuration to accept a type of `SecurityClientController` but when you read the value from the configuration it will be a `SecurityClient` perhaps it just needs two types even though they effectively look identical

Reply:
We agreed to decide to leave it how it is for the moment and see how the code feels. I want to help someone generating code to understand how it works.